lmfit/model.py

```diff
@@ -120,7 +170,8 @@ def propagate_err(z, dz, option):
 
         else:
             # Should never make it here, but don't want things to break
-            err = 0.0
+            # in a weird way, so be safe and raise exception
+            raise ValueError("Invalid parameter name ('%s') for function 'propagate_err'."%option)
```

Comment:
You now added an ```assert``` statement at the beginning of this function, which is good. As a consequence you're actually absolutely sure that it will never make it to this ```else``` statement, so just remove it completely.

Reply:
I think that in this case I'd rather remove the assertion statement and rely on the raise. This is purely a philosophical desire to have every if/then logical series terminate in an 'else'. Thoughts?

Reply:
I don't see any reason for every `if` to have an explicit `else`.

Reply:
Well, in this case the return value is being set via the logic chain. If the logic chain terminates without a default, the return variable is undefined. So terminating with a raise fixes this by overruling what would otherwise be an undefined error with a well-defined exception. Like I said, its philosophical. If the consensus is that I should do an explicit test/raise at the beginning of the function instead, I'm happy to do that both here and as a replacement for the remaining assert statement.